src/components/Microsoft.DotNet.UpgradeAssistant.Analysis/AnalyzeResultWriter.cs

```diff
@@ -51,40 +52,27 @@ await foreach (var ar in analyzeResults)
                             Name = ar.Name,
                             SemanticVersion = ar.Version,
                             InformationUri = ar.InformationURI,
-                            Rules = new List<ReportingDescriptor>()
+                            Rules = analyzeResults.GroupBy(x => x.RuleId).Select(a => new ReportingDescriptor()
                             {
-                                new()
-                                {
-                                    Id = ar.Id,
-                                    Name = ar.Name,
-                                },
-                            },
+                                Id = a.Key,
+                                FullDescription = new() { Text = a.First().RuleName, },
+                            }).ToList(),
                         },
                     },
-                    Results = await ExtractResultsAsync(ar).ConfigureAwait(false),
+                    Results = ExtractResults(analyzeResults),
                 };
             }
         }
 
-        private static async Task<IList<Result>> ExtractResultsAsync(AnalyzeResultDefinition result)
+        private static IList<Result> ExtractResults(IList<AnalyzeResult> analyzeResults)
```

Comment:
nit: do you need IList<>? Since this is private, can it just take List<>? Same for return type